uzsu/__init__.py

```diff
@@ -46,52 +46,98 @@
 
 from dateutil.rrule import rrulestr
 from dateutil import parser
 
 import lib.orb
 
-class UZSU():
+class UZSU(SmartPlugin):
+    """
+    Main class of the UZSU Plugin. Does all plugin specific stuff and provides
+    the update functions for the items
+    """
+    
+    ALLOW_MULTIINSTANCE = False
+    
+    PLUGIN_VERSION = "1.3.0"
+
 
     _items = {}         # item buffer for all uzsu enabled items
 
-    def __init__(self, smarthome, path=None):
+    def __init__(self, smarthome, path=None, *args, **kwargs):
+        """
+        Initalizes the plugin. The parameters describe for this method are pulled from the entry in plugin.conf.
+
+        :param sh:  The instance of the smarthome object, save it for later references
+        """
         self.logger = logging.getLogger('UZSU')
         self.logger.info('Init UZSU')
         self._sh = smarthome
 
-    def parse_item(self, item):
-        if 'uzsu_item' in item.conf:
-            self._items[item] = item()
-            return self.update_item
-
     def run(self):
-        """This is called once at the beginning after all items are already parsed from smarthome.py
+        """
+		This is called once at the beginning after all items are already parsed from smarthome.py
         All active uzsu items are registered to the scheduler
         """
+        self.logger.debug("run method called")
         self.alive = True
+        # if you want to create child threads, do not make them daemon = True!
+        # They will not shutdown properly. (It's a python bug)
+		
         for item in self._items:
             if 'active' in self._items[item]:
                 if self._items[item]['active']:
                     self._schedule(item)
 
     def stop(self):
+        """
+        Stop method for the plugin
+        """
+        self.logger.debug("stop method called")
         self.alive = False
 
+    def parse_item(self, item):
+        """
+        Default plugin parse_item method. Is called when the plugin is initialized.
+        The plugin can, corresponding to its attribute keywords, decide what to do with
+        the item in future, like adding it to an internal array for future reference
+
+        :param item:    The item to process.
+        :return:        If the plugin needs to be informed of an items change you should return a call back function
+                        like the function update_item down below. An example when this is needed is the knx plugin
+                        where parse_item returns the update_item function when the attribute knx_send is found.
+                        This means that when the items value is about to be updated, the call back function is called
+                        with the item, caller, source and dest as arguments and in case of the knx plugin the value
+                        can be sent to the knx with a knx write function within the knx plugin.
+
+        """
+	
+        if 'uzsu_item' in item.conf:
```

Comment:
Since it is single instance we don't need it right now but to make it more complete, we have to overwork the whole plugin. I just wanted to implement the most needed things...